inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn/runtime/debug_configuration.hpp"
+#include <iostream>
+
+namespace cldnn {
+
+const char *debug_configuration::prefix = "GPU_Debug: ";
+debug_configuration *debug_configuration::instance = nullptr;
+std::mutex debug_configuration::_m;
+
+// Default policy is that dump_configuration will override other configuration from IE.
+
+debug_configuration::debug_configuration()
+        : verbose(0)
+        , dump_graphs(std::string()) {
+#ifdef CLDNN_DEBUG_CONFIG
+    const std::string OV_GPU_VERBOSE("OV_GPU_Verbose");
+    const std::string OV_GPU_DUMP_GRAPHS("OV_GPU_DumpGraphs");
+    static std::once_flag flag;
+    std::string prefix_config = std::string(prefix) + "Config ";
+
+    if (const auto env_var = std::getenv(OV_GPU_VERBOSE.c_str())) {
+        verbose = std::stoi(env_var);
+        std::cout << prefix_config << OV_GPU_VERBOSE << "=" << verbose << std::endl;
+    }
+
+    if (const auto env_var = std::getenv(OV_GPU_DUMP_GRAPHS.c_str())) {
+        dump_graphs = env_var;
+        std::cout << prefix_config << OV_GPU_DUMP_GRAPHS << "=" << dump_graphs << std::endl;
+    }
+#endif
+}
+
+const debug_configuration *debug_configuration::get_instance() {
```

Comment:
why pointer? I think ref would be better. If debug is disabled, then we can probably throw an exception

Reply:
It is for the plain usage to call this function, as in https://github.com/openvinotoolkit/openvino/pull/6012/commits/52018792188cf369bd509f7c83a1e84bc5475ddb#diff-8b3f1cd1fffb5f10c598f8cf97aa15e3314d5d2cdeb060d6aa1d65cd3bc5ce73R181 .
However, it is possible to wrap this get_instance into the macro. I'll change it accordingly.

Reply:
Tried to make it as reference, --> need to remove get_instance() call in case DEBUG_CONFIG is not defined --> That causes compile failure in the following code where the code is trying to access member of debug_configuration.
So, at this moment, I think it is better to use pointer. Please let me know if you have any other idea.